# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1323045478 18000
#      Sun Dec 04 19:37:58 2011 -0500
# Node ID e58963b5c0b11d02febc7194d945e3398fc04f18
# Parent  e1f76bfe0452a6a9f789ca1a209e5cfb4616cf66
image.m: Fix test for non-linearly spaced coordinates.

diff --git a/scripts/image/image.m b/scripts/image/image.m
--- a/scripts/image/image.m
+++ b/scripts/image/image.m
@@ -91,18 +91,16 @@ endfunction
 ## variables are ignored.
 
 ## Author: Tony Richardson <arichard@stark.cc.oh.us>
 ## Created: July 1994
 ## Adapted-By: jwe
 
 function h = __img__ (x, y, img, varargin)
   
-  persistent warn = true
-
   newplot ();
 
   if (isempty (img))
     error ("__img__: matrix is empty");
   endif
 
   if (isempty (x))
     x = [1, columns(img)];
@@ -110,19 +108,23 @@ function h = __img__ (x, y, img, varargi
 
   if (isempty (y))
     y = [1, rows(img)];
   endif
 
   xdata = [x(1), x(end)];
   ydata = [y(1), y(end)];
 
-  if ((numel (unique (diff (x))) > 1 || numel (unique (diff (y))) > 1) && warn)
+  dx = diff (x);
+  dy = diff (y);
+  dx = std (dx) / mean (abs (dx));
+  dy = std (dy) / mean (abs (dy));
+  tol = 100*eps;
+  if (any (dx > tol) || any (dy > tol))
     warning ("Image does not map to non-linearly spaced coordinates")
-    warn = false;
   endif
 
   ca = gca ();
 
   tmp = __go_image__ (ca, "cdata", img, "xdata", xdata, "ydata", ydata,
                     "cdatamapping", "direct", varargin {:});
 
   px = __image_pixel_size__ (tmp);
